src/main/kotlin/no/nav/dagpenger/journalføring/arena/JournalføringArena.kt

```diff
@@ -30,19 +40,75 @@ class JournalføringArena() : Service() {
         val inngåendeJournalposter = builder.consumeTopic(INNGÅENDE_JOURNALPOST)
 
         inngåendeJournalposter
-                .peek { key, value -> LOGGER.info("Processing ${value.javaClass} with key $key") }
-                .filter { _, behov -> behov.getJournalpost().getJournalpostType() != null }
-                .filter { _, behov -> behov.getJournalpost().getBehandleneEnhet() != null }
-                .filter { _, behov -> behov.getJournalpost().getFagsakId() == null }
-                .mapValues(this::addFagsakId)
-                .peek { key, value -> LOGGER.info("Producing ${value.javaClass} with key $key") }
-                .toTopic(INNGÅENDE_JOURNALPOST)
+            .peek { key, value -> LOGGER.info("Processing ${value.javaClass} with key $key") }
+            .filter { _, behov -> behov.getJournalpost().getBehandleneEnhet() != null }
+            .filter { _, behov -> behov.getJournalpost().getFagsakId() == null }
+            .filter { _, behov -> behov.getJournalpost().getJournalpostType() != null }
+            .filter { _, behov -> filterJournalpostTypes(behov.getJournalpost().getJournalpostType()) }
+            .mapValues(this::addFagsakId)
+            .peek { key, value -> LOGGER.info("Producing ${value.javaClass} with key $key") }
+            .toTopic(INNGÅENDE_JOURNALPOST)
 
         return KafkaStreams(builder.build(), this.getConfig())
     }
 
+    override fun getConfig(): Properties {
+        return streamConfig(
+            appId = SERVICE_APP_ID,
+            bootStapServerUrl = env.bootstrapServersUrl,
+            credential = KafkaCredential(env.username, env.password)
+        )
+    }
+
+    private fun filterJournalpostTypes(journalpostType: JournalpostType): Boolean {
+        return when (journalpostType) {
+            NY, GJENOPPTAK, ETTERSENDING -> true
+            UKJENT, MANUELL -> false
+        }
+    }
+
     private fun addFagsakId(behov: Behov): Behov {
         val journalpost = behov.getJournalpost()
+
+        try {
+            val sakId = when (journalpost.getJournalpostType()) {
+                NY -> createNewSak(journalpost.getBehandleneEnhet(), journalpost.getSøker().getIdentifikator())
+                ETTERSENDING, GJENOPPTAK -> findSakAndCreateOppgave(
+                    journalpost.getBehandleneEnhet(),
+                    journalpost.getSøker().getIdentifikator()
+                )
+                else -> throw UnexpectedJournaltypeException("Unexpected journalposttype ${journalpost.getJournalpostType()}")
+            }
+
+            journalpost.setFagsakId(sakId)
+        } catch (oppslagException: OppslagException) {
```

Comment:
Perhaps we should not catch OppslagException? If we get a 404 or something similar from oppslag, we should not prosess any more journalposts until it's fixed